src/realm/sync/client_base.hpp

```diff
@@ -62,11 +62,11 @@ class SyncSocketProvider;
 struct ClientReset {
     realm::ClientResyncMode mode;
     DBRef fresh_copy;
-    bool recovery_is_allowed = true;
+    sync::ProtocolErrorInfo::Action action = sync::ProtocolErrorInfo::Action::ClientReset;
+    std::optional<Status> error;
```

Comment:
Yes, there is - I had made it optional since `Status` doesn't have a default constructor and `ClientReset` is default constructed and then populated throughout the code.

I could add an appropriate constructor and remove the `optional` if you prefer.

Reply:
why don't we just store the reason? I think that's all we need.

Reply:
It may be helpful to have the error code as well and it shouldn't be that expensive to store...

Reply:
right. it does feel a bit wrong to have the error optional though

Reply:
Updated the `ClientReset` structure so `error` is no longer an optional and updated the usages to use the initializers.